src/CommunicationApi/Services/Tablestorage/TableStorageBoxStore.cs

```diff
@@ -36,14 +36,23 @@ public async Task<string> Activate(string activationCode, string userName, strin
                 d.ActivationCode.Equals(activationCode, StringComparison.InvariantCultureIgnoreCase));
             if (device != null)
             {
-                device.Status = BoxStatus.Activated;
-                device.AdminUserName = userName;
-                device.AdminUserPhone = userPhone;
-                await Upsert(device, PartitionKey, device.BoxId);
+                // If status is already Activated, no need to update the device settings
+                if (device.Status != BoxStatus.Activated)
+                {
+                    device.Status = BoxStatus.Activated;
+                    device.AdminUserName = userName;
+                    device.AdminUserPhone = userPhone;
+                    await Upsert(device, PartitionKey, device.BoxId);
+                }
                 return device.BoxId;
             }
 
             return null;
         }
+
+        //public async Task<bool> IsAdmin(UserInfo userInfo)
```

Comment:
Can we remove this commented code please?

Reply:
done